backends/verilog/verilog_backend.cc

```diff
@@ -1983,14 +1983,64 @@ void dump_proc_switch(std::ostream &f, std::string indent, RTLIL::SwitchRule *sw
 		return;
 	}
 
+	bool is_cond = true;
+	for (auto it = sw->cases.begin(); it != sw->cases.end(); ++it) {
+		if ((*it)->compare.size() == 0) {
+			break;
+		} else if ((*it)->compare.size() == 1) {
+			int case_index = it - sw->cases.begin();
+			SigSpec compare = (*it)->compare.at(0);
+			if (case_index >= compare.size()) {
+				is_cond = false;
+				break;
+			}
+			if (compare[case_index] != State::S1)
+				is_cond = false;
+			for (int bit_index = 0; bit_index < compare.size(); bit_index++)
+				if (bit_index != case_index && compare[bit_index] != State::Sa)
+					is_cond = false;
+		} else is_cond = false;
+	}
+
+	if (is_cond) {
```

Comment:
Nit: I would prefer splitting this somehow. E.g. have a `dump_proc_switch_ifstring` returning false if the process cannot be handled the `if/else` way, in which case we would carry on with the `casez` emission in `dump_proc_switch`